tools/bwa.wdl

```diff
@@ -166,21 +166,23 @@ task bwa_aln_pe {
         tar -C bwa_db -xzf "~{bwa_db_tar_gz}" --no-same-owner
         PREFIX=$(basename bwa_db/*.ann ".ann")
 
-        ln -s "~{read_one_fastq_gz}" .
-        ln -s "~{read_two_fastq_gz}" .
+        CWD_R1='~{basename(read_one_fastq_gz)}'
+        CWD_R2='~{basename(read_two_fastq_gz)}'
+
+        ln -s "~{read_one_fastq_gz}" "$CWD_R1"
+        ln -s "~{read_two_fastq_gz}" "$CWD_R2"
 
         bwa sampe \
            -r "~{read_group}" \
             bwa_db/"$PREFIX" \
-            <(bwa aln -t "$n_cores" bwa_db/"$PREFIX" "~{basename(read_one_fastq_gz)}") \
-            <(bwa aln -t "$n_cores" bwa_db/"$PREFIX" "~{basename(read_two_fastq_gz)}") \
-            "~{basename(read_one_fastq_gz)}" "~{basename(read_two_fastq_gz)}" \
+            <(bwa aln -t "$n_cores" bwa_db/"$PREFIX" "$CWD_R1") \
+            <(bwa aln -t "$n_cores" bwa_db/"$PREFIX" "$CWD_R2") \
+            "$CWD_R1" "$CWD_R2" \
```

Comment:
@adthrasher this is passing the existing test, and I viewed the resulting `test.bam` and it checks out, but TBH don't understand what's going on in these 3 lines. Can you give me breakdown?

Reply:
Sure, the `aln` subcommand takes a single end read file and maps it into a `.sai` format file. Those are then passed to the `sampe` command to generate a paired-end alignment. These could be done sequentially:

```
bwa aln -t "$n_cores" bwa_db/"$PREFIX" "$CWD_R1" > sai1
bwa aln -t "$n_cores" bwa_db/"$PREFIX" "$CWD_R2" > sai2
bwa sampe -r "~{read_group}" bwa_db/"$PREFIX" sai1 sai2 "$CWD_R1" "$CWD_R2"
```

Instead, this uses process substitution (`<()`) to run the contained command (`bwa aln`) in a subshell and use the output as a temporary file/named pipe to the main command (`bwa sampe`). Since the `aln` commands are run in subshells, they can execute in parallel. Also the `sampe` command (I think) can begin as soon as the `aln` commands start writing output.